stat/sampleuv/withoutreplacement.go

```diff
@@ -0,0 +1,54 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sampleuv
+
+import (
+	"math/rand"
+	"sort"
+)
+
+// WithoutReplacement samples len(idxs) integers from [0, n) without replacement.
+// That is, upon return the elements of idxs will be unique integers. If source
```

Comment:
I'm afraid to ask in case I'm missing something obvious: if `n < len(idxs)`, can the elements of idxs be unique?